kadena/solidity/contracts/src/Wrapper.sol

```diff
@@ -0,0 +1,164 @@
+pragma solidity ^0.8.25;
+
+import {console} from "forge-std/Test.sol";
+import {SphinxVerifier as SphinxPlonkVerifier} from "sphinx-contracts/SphinxVerifier.sol";
+import "openzeppelin/access/Ownable.sol";
+
+struct SphinxProofFixture {
+    bytes proof;
+    bytes publicValues;
+    bytes32 vkey;
+}
+
+contract Wrapper is SphinxPlonkVerifier, Ownable(msg.sender) {
+    error ErrorUnexpectedLongestChainFixture();
+    error ErrorUnexpectedSpvFixture();
+    error ErrorUnexpectedSpvSubjectHash();
+    error ErrorConfirmationWorkIsSmallerThanThreshold();
+    error ErrorUnexpectedLayerHash();
+
+    // state
+    bytes32[] checkpoints;
+    uint256 confirmation_work_threshold;
+
+    constructor(uint256 confirmation_work_threshold_, bytes32[] memory checkpoints_) {
+        // Kadena's Chainweb protocol setting
+        require(checkpoints_.length == 5);
```

Comment:
Instead of forcing 5 as length we could check that the length is at least 5, as we don't have to prevent some user to want more security for the fallback in case of a fork.